src/cpp/src/generation_handle.cpp

```diff
@@ -20,11 +20,20 @@ bool GenerationHandleImpl::can_read() {
     return m_generation_stream->can_read();
 }
 
+bool GenerationHandleImpl::is_dropped() {
+    return get_status() == GenerationStatus::DROPPED_BY_HANDLE;
+}
+
+void GenerationHandleImpl::drop() {
+    m_generation_stream->drop();
+}
+
 std::unordered_map<uint64_t, GenerationOutput> GenerationHandleImpl::back() {
```

Comment:
I think we should block all methods not just `read`.

Reply:
done